### PR TITLE
upgrade to polkadot v1.3.0

### LINKED ISSUE #250: upgrade to polkadot-v1.3.0 and depend on crates.io
go from 1.0.0 directly to 1.3.0 which is the latest version [used by ORML](https://github.com/open-web3-stack/open-runtime-module-library/pull/963) and system chain runtimes as of today

also, remove redundancy in specifying dependencies by using
`pallet-balances = { workspace = true }`

### PR DESCRIPTION
**Pull request by brenzi**
closes #250 

design decision:
* Maintenance of node code is easier if we don't attempt to make service.rs generic. I think code redundancy is the lesser evil than maintenance effort here: we can simply copy-paste the template code and adjust two lines

* [x] must re-run benchmarks

### CODE REVIEW

**Review by clangenb — APPROVED** [review]
Nice, looks good to me

File: polkadot-parachains/integritee-runtime/src/weights/pallet_treasury.rs [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+//! DATE: 2023-12-15, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
 //! WORST CASE MAP SIZE: `1000000`
-//! HOSTNAME: `caribe`, CPU: `12th Gen Intel(R) Core(TM) i7-1260P`
-//! EXECUTION: Some(Wasm), WASM-EXECUTION: Compiled, CHAIN: Some("integritee-rococo-local-dev"), DB CACHE: 1024
+//! HOSTNAME: `ggwpez-ref-hw`, CPU: `Intel(R) Xeon(R) CPU @ 2.60GHz`
+//! WASM-EXECUTION: `Compiled`, CHAIN: `Some("../kusama-chain-spec.json")`, DB CACHE: 1024
 
 // Executed Command:
-// ./target/release/integritee-collator
+// ./target/production/polkadot
```
[the remark this answers is not in the document]
Comment (brenzi) [reply polkadot-parachains/integritee-runtime/src/weights/pallet_treasury.rs:26]: good point. noted as todo

File: polkadot-parachains/src/cli.rs [hunk 2]
```diff
@@ -39,24 +38,21 @@ pub enum Subcommand {
...
+   <bold>parachain-template-node build-spec --disable-default-bootnode > plain-parachain-chainspec.json</>
            Export a chainspec for a local testnet in json format.
-   <bold>integritee-collator --chain plain-parachain-chainspec.json --tmp -- --chain rococo-local</>
+   <bold>parachain-template-node --chain plain-parachain-chainspec.json --tmp -- --chain rococo-local</>
            Launch a full node with chain specification loaded from plain-parachain-chainspec.json.
-   <bold>integritee-collator</>
+   <bold>parachain-template-node</>
            Launch a full node with default parachain <italic>local-testnet</> and relay chain <italic>rococo-local</>.
-   <bold>integritee-collator --collator</>
+   <bold>parachain-template-node --collator</>
```
[the remark this answers is not in the document]
Comment (brenzi) [reply polkadot-parachains/src/cli.rs:55]: no. will fix

File: README.md [hunk 3]
```diff
@@ -76,13 +76,24 @@ The benchmarks are run with the following script:
 
 ```
 cargo build --release --features try-runtime
-./target/release/integritee-collator try-runtime \
-  --chain integritee-kusama \
-  --runtime ./target/release/wbuild/integritee-runtime/integritee_runtime.wasm \
-  on-runtime-upgrade --checks=all \
-  live --uri wss://kusama.api.integritee.network:443
+try-runtime --runtime ./target/release/wbuild/integritee-runtime/integritee_runtime.compact.compressed.wasm on-runtime-upgrade --checks=pre-and-post --disable-spec-version-check --disable-idempotency-checks live --uri wss://kusama.api.integritee.network:443
```
[the remark this answers is not in the document]
Comment (brenzi) [reply README.md:79]: hmmm, actually those test shouldn't hurt. true

File: Cargo.toml [hunk 4]
```diff
@@ -24,119 +24,120 @@ serde_json = "1.0.102"
...
+sc-transaction-pool = "25.0.0"
+sc-transaction-pool-api = "25.0.0"
+sp-blockchain = "25.0.0"
+sp-consensus = "0.29.0"
+sp-keyring = "28.0.0"
+sp-keystore = "0.31.0"
+sp-timestamp = "23.0.0"
+substrate-prometheus-endpoint = "0.16.0"
+substrate-state-trie-migration-rpc = "24.0.0"
+try-runtime-cli = "0.35.0"
```
Comment (clangenb) [inline Cargo.toml:140]: Isn't this a standalone binary now that we don't need anymore?

**Comment by brenzi** [comment]
hmm, runtime-benchmark build works, but 

```
./target/release/integritee-collator benchmark pallet --list --chain integritee-rococo-local-dev
Error: Input("Did not find the benchmarking metadata. This could mean that you either did not build the node correctly with the `--features runtime-benchmarks` flag, or the chain spec that you are using was not created by a node that was compiled with the flag: Other: Exported method Benchmark_benchmark_metadata is not found")
```

**Comment by brenzi** [comment]
for the sake of completeness, I tested that the parachain template in polkadot-sdk works correctly:

```
cargo build --release --features runtime-benchmarks -p parachain-template-node
./target/release/parachain-template-node benchmark pallet --list --chain dev
```

works as expected

**Pull request merged**